### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by mihaic**
See PyTest docs:
https://docs.pytest.org/en/4.6-maintenance/deprecations.html#message-parameter-of-pytest-raises

And NumPy bug:
https://github.com/numpy/numpy/issues/14189

### CODE REVIEW

**Comment by mihaic** [comment]
@brainiak/committers, please review.

The changes are unrelated to each other, but all necessary to have the tests pass again.

**Comment by mihaic** [comment]
A couple of references for the `conda-build` changes:
https://conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#macos-sdk
https://conda.io/projects/conda-build/en/latest/resources/variants.html#conda-variables-and-command-line-arguments-to-conda-build

The SciPy pinning is the same as in `setup.py`.

**Comment by mihaic** [comment]
@danielsuo, @gdoubleyew, any chance you could review?

File: .conda/build.sh [hunk 1]
```diff
@@ -1,13 +1,5 @@
 #!/bin/bash
 
-if [[ "$OSTYPE" == "darwin"* ]]; then
```
Comment (gdoubleyew) [inline .conda/build.sh]: Why did this block go away?

Comment (mihaic) [reply .conda/build.sh]: `MACOSX_DEPLOYMENT_TARGET` is set by `conda-build` based on the SDK we pass. The compiler variables are not used; instead, `conda-build` installs and uses compilers based on its configuration.

File: tests/eventseg/test_event.py [hunk 2]
```diff
@@ -33,10 +33,12 @@ def test_fit_shapes():
         "Segmentation from find_events not correctly normalized"
 
     es_invalid = EventSegment(K)
-    with pytest.raises(ValueError, message="T < K should cause error"):
+    with pytest.raises(ValueError):
         es_invalid.model_prior(K-1)
-    with pytest.raises(ValueError, message="#Events < K should cause error"):
+        pytest.fail("T < K should cause error")
```
[the remark this answers is not in the document]
Comment (mihaic) [reply tests/eventseg/test_event.py]: Done.

**Pull request merged**